botocore/session.py

```diff
@@ -984,6 +987,7 @@ def create_client(
                     ignored_credentials,
                 )
             credentials = self.get_credentials()
+        register_feature_id('CREDENTIALS_CODE')
```

Comment:
As the code is today, this will register it for all credentials, regardless of source.  I should have been clearer in my previous comment, the if statement is still required.  I have two potential suggestions here with no strong preference between them.  We can either:

1. Avoid the if statement by registering this in two places: 
    - once in this method on line 927 where we set credentials to a `botocore.credentials.Credentials` object
    - once in the [set_credentials method ](https://github.com/boto/botocore/blob/develop/botocore/session.py#L484-L508)that [boto3 uses](https://github.com/boto/boto3/blob/develop/boto3/session.py#L98) when credentials are set on a session.
2. Keep this here, but add back the if statement.  Since that's easy enough for me to add as a suggestion, I've done that below.

```suggestion
        if getattr(self._credentials, 'method', None) == 'explicit':
            register_feature_id('CREDENTIALS_CODE')
```


Please consider the pros and cons of each and pick the one you prefer (or a third option I'm not considering)

Reply:
I decided to go with option 2. Also, I am assuming its a typo and you meant to write `if getattr(credentials, 'method', None) == 'explicit':`